registry/auth.go

```diff
@@ -271,14 +269,10 @@ func Login(authConfig *AuthConfig, factory *utils.HTTPRequestFactory) (string, e
 }
 
 // this method matches a auth configuration to a server address or a url
-func (config *ConfigFile) ResolveAuthConfig(hostname string) AuthConfig {
-	if hostname == IndexServerAddress() || len(hostname) == 0 {
-		// default to the index server
-		return config.Configs[IndexServerAddress()]
-	}
-
+func (config *ConfigFile) ResolveAuthConfig(index *IndexInfo) AuthConfig {
+	configKey := index.GetAuthConfigKey()
 	// First try the happy case
-	if c, found := config.Configs[hostname]; found {
+	if c, found := config.Configs[configKey]; found || index.Official {
```

Comment:
@dkjer why is there `|| index.Official` ? That would mean that if the config was not found, it would return an empty `AuthConfig`. Is that what we want? Is that what the old code did?